test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ConditionalCreateTests.cs

```diff
@@ -104,5 +110,52 @@ public async Task GivenAResource_WhenCreatingConditionallyWithEmptyIfNoneHeader_
 
             Assert.Equal(HttpStatusCode.BadRequest, exception.Response.StatusCode);
         }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenConcurrentResources_WhenCreatingConditionallyWithNoIdAndNoExisting_TheServerShouldReturnOneResourceSuccessfully()
+        {
+            string identifier = Guid.NewGuid().ToString();
+            string value = Guid.NewGuid().ToString();
+
+            int count = 0;
+
+            var observation = Samples.GetDefaultObservation().ToPoco<Observation>();
+            observation.Id = null;
+            observation.Identifier.Add(new Identifier(null, identifier));
+            observation.Value = new FhirString(value);
+
+            async Task<HttpStatusCode> ExecuteRequest()
+            {
+                try
+                {
+                    // Both criteria should link to the same resource, but with different queries
+                    string criteria = $"identifier={identifier}";
+
+                    if (Interlocked.Increment(ref count) % 2 == 0)
+                    {
+                        criteria = $"value-string={value}";
+                    }
+
+                    var response = await _client.CreateAsync(observation, criteria);
+                    return response.StatusCode;
+                }
+                catch (FhirException ex)
+                {
+                    return ex.StatusCode;
+                }
+            }
+
+            HttpStatusCode[] tasks = await Task.WhenAll(Enumerable.Range(1, 30)
+                .Select(_ => ExecuteRequest()));
+
+            foreach (var item in tasks)
+            {
+                _logger.WriteLine("Status:" + item);
+            }
+
+            Assert.Equal(1, tasks.Count(x => x == HttpStatusCode.Created));
+            Assert.Equal(tasks.Length - 1, tasks.Count(x => x == HttpStatusCode.Conflict || x == HttpStatusCode.OK || x == HttpStatusCode.PreconditionFailed));
```

Comment:
>HttpStatusCode.OK  [](start = 97, length = 18)

What `Ok` suppose to represent?

Reply:
`PreconditionFailed` concerns me more.

- OK: The query found a result, so nothing would be created, operation returns OK.
- Conflict: From the revised logic in this PR, two items were created at the same time.
- PreconditionFailed: A thread executed the search and found both the created item and an item being reverted.